src/components/main.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
+
+import friendListData from '../data/userdata.json';
+
+export default class Main extends React.Component {
+    constructor(props){
+        super(props)
+        this.state = {
+            friendList : friendListData,
+            searchFriendName : ''
+        }
+    }
+    render(){
+   
+        const printFriends = (friends) => {
+            friends = friends.filter((friend) => {
+                return friend.name.indexOf(this.state.searchFriendName) > -1;
+            });
+            return friends.map((friend) => {
+                return (
+                    <Link to={{
+                        pathname: `/chatting/${friend.id}`,
+                        state: {
+                            id : friend.id
+                        },
+                        }}
+                    >
+                        <Profile>
+                            <ProfileImg src={friend.image} alt={friend.name}></ProfileImg>
+                            <ProfileDesc>
+                                    <ProfileName>{friend.name}</ProfileName>
+                                    <ProfileState>{friend.state}</ProfileState>
+                            </ProfileDesc>
+                        </Profile>
+                    </Link>
+                )
+            })
+        }
+        return(
+                <Wrapper>
+                    <Title>react 메신저</Title>
+                
+                    <MyProfile>
+                        
+                        <ProfileDesc>
+                            <ProfileName>integer</ProfileName>
+                            <ProfileState>테스트입니다.</ProfileState>
+                        </ProfileDesc>
+                    </MyProfile>
+                    {printFriends(this.state.friendList.friendList)}
```

Comment:
```suggestion
                    <FriendList list={this.state.friendList.friendList}/>
```
이런식으로 JSX 내에선 JSX가 나오는게 아무래도 가독성에선 유리하다고 생각해요.